### ISSUE TITLE
Try out msgpack-java v0.7 support for nested maps and arrays

### ISSUE THREAD

**Issue by poros** [Yelp/pyleus#96]
Support for nested maps and arrays has been implemented in msgpaxk-java v0.7.x (https://github.com/msgpack/msgpack-java/issues/131)

We ended up writing that behavior by ourselves in the java serializer, so we are kinda fine with that for now, but if this work as intended it may make our life a little easier. Definitely worth a try.

**Comment by jandk** [comment]
Yes, we are running into the same limitation. We need to switch to msgpack-core, because we need explicit String support.
